src/commands/cli/release/build.ts

```diff
@@ -0,0 +1,239 @@
+      char: 'g',
+      exactlyOne: ['start-from-npm-dist-tag', 'start-from-github-ref'],
+    }),
+    'build-only': Flags.boolean({
+      summary: messages.getMessage('flags.buildOnly'),
+      default: false,
+    }),
+    resolutions: Flags.boolean({
+      summary: messages.getMessage('flags.resolutions'),
+      default: true,
+      allowNo: true,
+    }),
+    only: arrayWithDeprecation({
+      summary: messages.getMessage('flags.only'),
+    }),
+    'pinned-deps': Flags.boolean({
+      summary: messages.getMessage('flags.pinnedDeps'),
+      default: true,
+      allowNo: true,
+    }),
+    patch: Flags.boolean({
+      summary: messages.getMessage('flags.patch'),
+      exclusive: ['prerelease'],
+    }),
+    prerelease: Flags.string({
+      summary: messages.getMessage('flags.prerelease'),
+      exclusive: ['patch'],
+    }),
+    snapshot: Flags.boolean({
+      summary: messages.getMessage('flags.snapshot'),
+    }),
+    schema: Flags.boolean({
+      summary: messages.getMessage('flags.schema'),
+    }),
+  };
+
+  public async run(): Promise<void> {
+    // I could not get the shelljs.exec config { fatal: true } to actually throw an error, but this works :shrug:
+    set('-e');
+
+    let auth: string;
+    const { flags } = await this.parse(build);
+
+    if (flags.prerelease === 'true' || flags.prerelease === 'false') {
+      throw new SfError(
+        'The prerelease flag is not a boolean. It should be the name of the prerelease tag, examples: dev, alpha, beta'
+      );
+    }
+
+    const pushChangesToGitHub = !flags['build-only'];
+
+    if (pushChangesToGitHub) {
+      auth = ensureString(
+        new Env().getString('GH_TOKEN') ?? new Env().getString('GITHUB_TOKEN'),
+        'The GH_TOKEN env var is required to push changes to GitHub. Use the --build-only flag to skip GitHub operations (a manual push will then be needed)'
+      );
+    }
+
+    const { ['start-from-npm-dist-tag']: startFromNpmDistTag, ['start-from-github-ref']: startFromGithubRef } = flags;
+
+    let ref: string;
+
+    if (startFromGithubRef) {
+      this.log(`Flag '--start-from-github-ref' passed, switching to '${startFromGithubRef}'`);
+
+      ref = startFromGithubRef;
+    } else {
+      this.log(`Flag '--start-from-npm-dist-tag' passed, looking up version for ${startFromNpmDistTag}`);
+
+      // Classes... I wish this was just a helper function.
+      const temp = await PackageRepo.create({ ux: new Ux({ jsonEnabled: this.jsonEnabled() }) });
+      const version = temp.package.getDistTags(temp.package.packageJson.name)[startFromNpmDistTag];
+
+      ref = version;
+    }
+
+    // Check out "starting point"
+    // Works with sha (detached): "git checkout f476e8e"
+    // Works with remote branch:  "git checkout my-branch"
+    // Works with tag (detached): "git checkout 7.174.0"
+    this.exec(`git checkout ${ref}`);
+
+    const repo = await PackageRepo.create({ ux: new Ux({ jsonEnabled: this.jsonEnabled() }) });
```

Comment:
if you're going to create the package down here anyway, you could have used it for the `startFromNpmDistTag` stuff instead of creating it there, too

Reply:
This is sort of a chicken and egg situation. In the `getGithubRef` method, I need to look up the github tag version for the passed in npm dist-tag before we can check it out. If we create a `PackageRepo` first before checking out the dist-tag version, all of the class's properties would be incorrect. Like I had mentioned in the comment, it would be nice if that was just a helper function. I was torn between creating a temp Package and just calling `npm view`.